BUG: Build failure on PyPy3.10 7.3.16: `error: ‘Py_Initialize’ was not declared in this scope`

### Describe your issue.

When building scipy on PyPy3.10 7.3.16, the build fails because of undeclared `Py_Initialize` function. I've reproduced this with 19dcd9fbaaf99cf0020f050b82b7054ef5dbccf6. FWICS none of PyPy headers declare this one.

CC @mattip, @cfbolz

### Reproducing Code Example

```python
# install PyPy3.10 7.3.16
pypy3 -m build -w
```


### Error message

```shell
[850/1383] Compiling C++ object scipy/special/_gufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_gufuncs.cpp.o
FAILED: scipy/special/_gufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_gufuncs.cpp.o 
ccache c++ -Iscipy/special/_gufuncs.pypy310-pp73-x86_64-linux-gnu.so.p -Iscipy/special -I../scipy/special -Iscipy/_lib -I../scipy/_lib -I../scipy/_build_utils/src -I../../build-env-9o3hi5jk/lib/pypy3.10/site-packages/numpy/_core/include -I/usr/include/pypy3.10 -fvisibility=hidden -fvisibility-inlines-hidden -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++17 -O3 -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION -DSP_SPECFUN_ERROR -MD -MQ scipy/special/_gufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_gufuncs.cpp.o -MF scipy/special/_gufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_gufuncs.cpp.o.d -o scipy/special/_gufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_gufuncs.cpp.o -c ../scipy/special/_gufuncs.cpp
In file included from ../scipy/special/_gufuncs.cpp:1:
../scipy/special/ufunc.h: In function ‘bool SpecFun_Initialize()’:
../scipy/special/ufunc.h:22:5: error: ‘Py_Initialize’ was not declared in this scope; did you mean ‘Py_IsInitialized’?
   22 |     Py_Initialize();
      |     ^~~~~~~~~~~~~
      |     Py_IsInitialized
[853/1383] Compiling C++ object scipy/special/_special_ufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_special_ufuncs.cpp.o
FAILED: scipy/special/_special_ufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_special_ufuncs.cpp.o 
ccache c++ -Iscipy/special/_special_ufuncs.pypy310-pp73-x86_64-linux-gnu.so.p -Iscipy/special -I../scipy/special -Iscipy/_lib -I../scipy/_lib -I../scipy/_build_utils/src -I../../build-env-9o3hi5jk/lib/pypy3.10/site-packages/numpy/_core/include -I/usr/include/pypy3.10 -fvisibility=hidden -fvisibility-inlines-hidden -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++17 -O3 -fPIC -DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION -DSP_SPECFUN_ERROR -MD -MQ scipy/special/_special_ufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_special_ufuncs.cpp.o -MF scipy/special/_special_ufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_special_ufuncs.cpp.o.d -o scipy/special/_special_ufuncs.pypy310-pp73-x86_64-linux-gnu.so.p/_special_ufuncs.cpp.o -c ../scipy/special/_special_ufuncs.cpp
In file included from ../scipy/special/_special_ufuncs.cpp:1:
../scipy/special/ufunc.h: In function ‘bool SpecFun_Initialize()’:
../scipy/special/ufunc.h:22:5: error: ‘Py_Initialize’ was not declared in this scope; did you mean ‘Py_IsInitialized’?
   22 |     Py_Initialize();
      |     ^~~~~~~~~~~~~
      |     Py_IsInitialized
```


### SciPy/NumPy/Python version and system information

```shell
n/a (it didn't build)
```

--- a/scipy/special/ufunc.h
+++ b/scipy/special/ufunc.h
@@ -17,10 +17,8 @@
 #include "sf_error.h"
 #include "special/mdspan.h"
 
-// Initializes Python and NumPy.
+// Initializes NumPy.
 inline bool SpecFun_Initialize() {
-    Py_Initialize();
-
     import_array();
     if (PyErr_Occurred() != nullptr) {
         return false; // import array failed

